client/src/pages/Board.js

```diff
@@ -35,20 +37,59 @@ const useStyles = makeStyles((theme) => ({
       borderColor: "#ffffff",
     },
   },
+  dropdown: {
+    minWidth: "100px",
+    padding: "1rem",
+    background: "#ffffff",
+    zIndex: "1",
+    position: "absolute",
+    top: "160px",
+    right: "1rem",
+    boxShadow: "0px 0px 10px 1px rgba(0,0,0,0.15)",
+    textAlign: "right",
+  },
 }));
 
-function Board() {
+function Board(props) {
   const classes = useStyles();
-  const { user } = useContext(UserContext);
+  const { boardList, currBoardId, setCurrBoardId, currBoard } = useContext(
+    UserContext
+  );
   const { socket } = useContext(SocketContext);
   const [showModal, setShowModal] = useState(false);
   const [showUpload, setShowUpload] = useState(false);
+  const [showDropdown, setShowDropdown] = useState(false);
+  const { id } = useParams();
+
+  useEffect(() => {
+    setCurrBoardId(id);
+  }, [id, setCurrBoardId]);
+  useEffect(() => {
+    if (currBoardId) {
+      props.history.push(`/board/${currBoardId}`);
```

Comment:
I don't know if you guys have been using props.history.push in other parts of the code, but if you usually use `react-router-dom` components, it's better to be consistent and stick with those instead of mixing them with `history` functions. Eg. have a `redirect` state and if it's true, redirect to `/board/${currBoardId}` from render